internal/state/relationship/capturer_test.go

```diff
@@ -0,0 +1,304 @@
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/helpers"
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/relationship"
+	"github.com/nginxinc/nginx-kubernetes-gateway/pkg/sdk"
+)
+
+func createBackendRefs(backendNames ...v1beta1.ObjectName) []v1beta1.HTTPBackendRef {
+	refs := make([]v1beta1.HTTPBackendRef, 0, len(backendNames))
+	for _, name := range backendNames {
+		refs = append(refs, v1beta1.HTTPBackendRef{
+			BackendRef: v1beta1.BackendRef{
+				BackendObjectReference: v1beta1.BackendObjectReference{
+					Kind:      (*v1beta1.Kind)(helpers.GetStringPointer("Service")),
+					Name:      name,
+					Namespace: (*v1beta1.Namespace)(helpers.GetStringPointer("test")),
+				},
+			},
+		})
+	}
+
+	return refs
+}
+
+func createRules(backendRefs ...[]v1beta1.HTTPBackendRef) []v1beta1.HTTPRouteRule {
+	rules := make([]v1beta1.HTTPRouteRule, 0, len(backendRefs))
+	for _, refs := range backendRefs {
+		rules = append(rules, v1beta1.HTTPRouteRule{BackendRefs: refs})
+	}
+
+	return rules
+}
+
+func createRoute(name string, rules []v1beta1.HTTPRouteRule) *v1beta1.HTTPRoute {
+	return &v1beta1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{Namespace: "test", Name: name},
+		Spec:       v1beta1.HTTPRouteSpec{Rules: rules},
+	}
+}
+
+var _ = Describe("Capturer", func() {
+	var (
+		capturer *relationship.Capturer
+
+		backendRef1 = createBackendRefs("svc1")
+		backendRef2 = createBackendRefs("svc2")
+		backendRef3 = createBackendRefs("svc3")
+		backendRef4 = createBackendRefs("svc4")
+
+		hr1 = createRoute("hr1", createRules(backendRef1))
+		hr2 = createRoute("hr2", createRules(backendRef2, backendRef3, backendRef4))
+
+		hrSvc1AndSvc2 = createRoute("hr-svc1-svc2", createRules(backendRef1, backendRef2))
+		hrSvc1AndSvc3 = createRoute("hr-svc1-svc3", createRules(backendRef3, backendRef1))
+		hrSvc1AndSvc4 = createRoute("hr-svc1-svc4", createRules(backendRef1, backendRef4))
+
+		hr1Name           = types.NamespacedName{Namespace: hr1.Namespace, Name: hr1.Name}
+		hr2Name           = types.NamespacedName{Namespace: hr2.Namespace, Name: hr2.Name}
+		hrSvc1AndSvc2Name = types.NamespacedName{Namespace: hrSvc1AndSvc2.Namespace, Name: hrSvc1AndSvc2.Name}
+		hrSvc1AndSvc3Name = types.NamespacedName{Namespace: hrSvc1AndSvc3.Namespace, Name: hrSvc1AndSvc3.Name}
+		hrSvc1AndSvc4Name = types.NamespacedName{Namespace: hrSvc1AndSvc4.Namespace, Name: hrSvc1AndSvc4.Name}
+
+		svc1 = types.NamespacedName{Namespace: "test", Name: "svc1"}
+		svc2 = types.NamespacedName{Namespace: "test", Name: "svc2"}
+		svc3 = types.NamespacedName{Namespace: "test", Name: "svc3"}
+		svc4 = types.NamespacedName{Namespace: "test", Name: "svc4"}
+	)
+
+	Describe("Capture service relationships for routes", func() {
+		BeforeEach(OncePerOrdered, func() {
+			capturer = relationship.NewCapturer()
+		})
+		Describe("Normal cases", Ordered, func() {
+			When("a route with a backend service is captured", func() {
```

Comment:
using when makes it much more readable 👍  I'll make sure to use it in my future tests